src/Microsoft.ML.Tokenizers/Tokenizer.cs

```diff
@@ -136,6 +137,70 @@ public int CountTokens(string text, bool considerSpecialTokens = true)
             return idsCount;
         }
 
+        /// <summary>
+        /// Find the index of the maximum encoding capacity from the start within the text without surpassing the token limit.
+        /// </summary>
+        /// <param name="text">The text to encode.</param>
+        /// <param name="maxTokenCount">The maximum token count to limit the encoding capacity.</param>
+        /// <param name="processedText">If the tokenizer's normalization is enabled, the input text will be represented in its normalization form; otherwise, it will remain unchanged as the input text.</param>
+        /// <param name="tokenCount">The token count can be generated which should be smaller than the maximum token count.</param>
+        /// <param name="considerSpecialTokens">Indicate if want to consider the special tokens during the encoding.</param>
+        /// <returns>The index of the maximum encoding capacity within the processed text without surpassing the token limit.</returns>
```

Comment:
I think it'd be worth being very specific about what this means. It's the index just after the last character to be included, which means if no tokens fit, you'll get back 0, and if all tokens fit, you'll get back text.Length.